### PR TITLE
Use inspectField to simplify code

### PR DESCRIPTION
**Pull request by hqhq**
Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

File: integration-cli/docker_cli_inspect_test.go [hunk 1]
```diff
@@ -12,13 +12,12 @@ import (
 func (s *DockerSuite) TestInspectImage(c *check.C) {
 	imageTest := "emptyfs"
 	imageTestID := "511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158"
-	imagesCmd := exec.Command(dockerBinary, "inspect", "--format='{{.Id}}'", imageTest)
-	out, exitCode, err := runCommandWithOutput(imagesCmd)
-	if exitCode != 0 || err != nil {
-		c.Fatalf("failed to inspect image: %s, %v", out, err)
+	id, err := inspectField(imageTest, "Id")
+	if err != nil {
```
Comment (runcom) [inline integration-cli/docker_cli_inspect_test.go]: `c.Assert(err, check.IsNil)`

File: integration-cli/docker_cli_inspect_test.go [hunk 2]
```diff
@@ -33,33 +32,31 @@ func (s *DockerSuite) TestInspectInt64(c *check.C) {
 
 	out = strings.TrimSpace(out)
 
-	inspectCmd := exec.Command(dockerBinary, "inspect", "-f", "{{.HostConfig.Memory}}", out)
-	inspectOut, _, err := runCommandWithOutput(inspectCmd)
+	inspectOut, err := inspectField(out, "HostConfig.Memory")
 	if err != nil {
```
Comment (runcom) [inline integration-cli/docker_cli_inspect_test.go]: `c.Assert` as said above, pls everywhere a trivial err check is done :)

File: integration-cli/docker_cli_run_test.go [hunk 3]
```diff
@@ -101,12 +101,9 @@ func (s *DockerSuite) TestRunEchoStdoutWitCPUQuota(c *check.C) {
 		c.Errorf("container should've printed 'test'")
 	}
 
-	cmd := exec.Command(dockerBinary, "inspect", "-f", "{{.HostConfig.CpuQuota}}", "test")
-	out, _, err = runCommandWithOutput(cmd)
-	if err != nil {
-		c.Fatalf("failed to inspect container: %s, %v", out, err)
+	if out, err = inspectField("test", "HostConfig.CpuQuota"); err != nil {
+		c.Errorf("%v", err)
```
Comment (runcom) [inline integration-cli/docker_cli_run_test.go]: why not fatal?

**Comment by runcom** [comment]
@hqhq I may be wrong but seems not all `dockerBinary, "inspect"` have been removed? could you try to grep for `dockerBinary, "inspect"` and see if you catched everything? :)

File: integration-cli/docker_cli_inspect_test.go (hunk 2 above)
Comment (hqhq) [inline integration-cli/docker_cli_inspect_test.go]: Yeah, thanks for the tip, I'll change them.

File: integration-cli/docker_cli_run_test.go (hunk 3 above)
Comment (hqhq) [inline integration-cli/docker_cli_run_test.go]: My mistake, but seem there are lots of `c.Errorf` in that file, maybe see it in another PR.

**Comment by hqhq** [comment]
@runcom I use:
`grep -rnI "\"inspect\"" integration-cli`
and changed all that I can see is safe. Some exceptions:
 - use other format like {{ len xxx }} or {{ range xxx }} which are not wrapped by a function
 - with .CombinedOutput in the tail and I haven't see if it's safe to directly change
 - in daemon test and they are used in different way

**Comment by hqhq** [comment]
Updated.

**Comment by runcom** [comment]
@hqhq thanks, LGTM

**Comment by cpuguy83** [comment]
LGTM

File: integration-cli/docker_cli_start_test.go [hunk 4]
```diff
@@ -260,12 +246,10 @@ func (s *DockerSuite) TestStartAttachMultipleContainers(c *check.C) {
 
 	// confirm the state of all the containers be stopped
 	for container, expected := range map[string]string{"test1": "false", "test2": "false", "test3": "false"} {
-		cmd = exec.Command(dockerBinary, "inspect", "-f", "{{.State.Running}}", container)
-		out, _, err := runCommandWithOutput(cmd)
+		out, err := inspectField(container, "State.Running")
 		if err != nil {
```
Comment (cpuguy83) [inline integration-cli/docker_cli_start_test.go]: Missed on assert! :)

**Pull request merged**